lib/common/variadic/uint32OrHash.go

```diff
@@ -97,37 +97,62 @@ func (x *Uint32OrHash) Hash() common.Hash {
 	return x.value.(common.Hash)
 }
 
+func (x *Uint32OrHash) String() string {
+	if x.IsHash() {
+		return x.Hash().String()
+	}
+
+	return fmt.Sprintf("%d", x.value)
+}
+
 // IsUint32 returns true if the value is a uint32
 func (x *Uint32OrHash) IsUint32() bool {
-	if x == nil {
+	switch x.Value().(type) {
+	case int, uint, uint32:
+		return true
+	default:
 		return false
 	}
-
-	_, is := x.value.(uint32)
-	return is
 }
 
 // Uint32 returns the value as a uint32. It panics if the value is not a uint32.
 func (x *Uint32OrHash) Uint32() uint32 {
-	if !x.IsUint32() {
+	var blockNumber uint32
+
+	switch c := x.Value().(type) {
+	case uint32:
+		blockNumber = c
+	case int:
+		blockNumber = uint32(c)
+	case uint:
+		blockNumber = uint32(c)
```

Comment:
shouldn't you check for overflow?

Reply:
I am wondering if we just use uint instead of uint32 since that is the type used for block number? @timwu20 wdyt?

Reply:
created PR that solves the variadic check problem https://github.com/ChainSafe/gossamer/pull/4191